### PR TITLE
update getobjectbyobjectid api version to 1.6 from 1.6-internal

### LINKED ISSUE #3407: Get-AzureRmRoleAssignment command failing
### PowerShell Version
Instructions: to get PowerShell version, type `$PSVersionTable` and look for the value associated with `PSVersion`
latest
### Module Version
Azure PS version: 1.0.0
### OS Version
Instructions: to get OS version, type `$PSversionTable` and look for value associated with `BuildVersion`
Windows 2012 R2
### Description
Error when trying to run the PS command:
Get-AzureRmRoleAssignment 
: ##[error]Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
2017-01-18T08:40:40.1990848Z Microsoft.Azure.Commands.Resources.Models.Authorization.PSRoleDefinition
### Debug Output
Instructions: to get Debug Output, set `$DebugPreference="Continue"` and then execute the cmdlet or script causing the issue

### Script/Steps for Reproduction
2017-01-18T17:43:54.3907947Z ##[debug]Leaving Initialize-Azure.
2017-01-18T17:43:54.4499767Z ##[command]& 'C:\Users\buildguest\AppData\Local\Temp\bfdd2b32-1828-48c5-a451-d7143f0a9e30.ps1' 
2017-01-18T17:43:56.0065186Z ##[debug]Error record:
2017-01-18T17:43:56.1317540Z ##[debug]Get-AzureRmRoleAssignment : Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
2017-01-18T17:43:56.1337554Z ##[debug]At C:\Users\buildguest\AppData\Local\Temp\bfdd2b32-1828-48c5-a451-d7143f0a9e30.ps1:3 char:1
2017-01-18T17:43:56.1357540Z ##[debug]+ Get-AzureRmRoleAssignment -verbose
2017-01-18T17:43:56.1367563Z ##[debug]+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2017-01-18T17:43:56.1387550Z ##[debug]    + CategoryInfo          : CloseError: (:) [Get-AzureRmRoleAssignment], CloudException
2017-01-18T17:43:56.1407543Z ##[debug]    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Resources.GetAzureRoleAssignmentCommand
2017-01-18T17:43:56.1417557Z ##[debug] 
2017-01-18T17:43:56.1447543Z ##[debug]Script stack trace:
2017-01-18T17:43:56.1497562Z ##[debug]at <ScriptBlock>, C:\Users\buildguest\AppData\Local\Temp\bfdd2b32-1828-48c5-a451-d7143f0a9e30.ps1: line 3
2017-01-18T17:43:56.1517543Z ##[debug]at <ScriptBlock>, <No file>: line 1
2017-01-18T17:43:56.1527562Z ##[debug]at <ScriptBlock>, C:\a\_tasks\AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62\1.1.5\AzurePowerShell.ps1: line 64
2017-01-18T17:43:56.1547543Z ##[debug]at <ScriptBlock>, C:\a\_tasks\AzurePowerShell_72a1931b-effb-4d2e-8fd8-f8472a07cb62\1.1.5\AzurePowerShell.ps1: line 60
2017-01-18T17:43:56.1557558Z ##[debug]at <ScriptBlock>, <No file>: line 1
2017-01-18T17:43:56.1577601Z ##[debug]at <ScriptBlock>, <No file>: line 22
2017-01-18T17:43:56.1597546Z ##[debug]at <ScriptBlock>, <No file>: line 18
2017-01-18T17:43:56.1613921Z ##[debug]at <ScriptBlock>, <No file>: line 1
2017-01-18T17:43:56.1644002Z ##[debug]Exception:
2017-01-18T17:43:56.2355650Z ##[debug]Microsoft.Rest.Azure.CloudException: Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
2017-01-18T17:43:56.2375685Z ##[debug]   at Microsoft.Azure.Graph.RBAC.ObjectsOperations.<GetObjectsByObjectIdsWithHttpMessagesAsync>d__6.MoveNext()
2017-01-18T17:43:56.2395630Z ##[debug]--- End of stack trace from previous location where exception was thrown ---
2017-01-18T17:43:56.2415644Z ##[debug]   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2017-01-18T17:43:56.2425679Z ##[debug]   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-01-18T17:43:56.2445651Z ##[debug]   at Microsoft.Azure.Graph.RBAC.ObjectsOperationsExtensions.<GetObjectsByObjectIdsAsync>d__3.MoveNext()
2017-01-18T17:43:56.2465644Z ##[debug]--- End of stack trace from previous location where exception was thrown ---
2017-01-18T17:43:56.2485657Z ##[debug]   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
2017-01-18T17:43:56.2495653Z ##[debug]   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-01-18T17:43:56.2516192Z ##[debug]   at Microsoft.Azure.Graph.RBAC.ObjectsOperationsExtensions.GetObjectsByObjectIds(IObjectsOperations operations, GetObjectsParameters parameters)
2017-01-18T17:43:56.2536390Z ##[debug]   at Microsoft.Azure.Commands.Resources.Models.ActiveDirectory.ActiveDirectoryClient.GetObjectsByObjectId(List`1 objectIds)
2017-01-18T17:43:56.2549680Z ##[debug]   at Microsoft.Azure.Commands.Resources.Models.Authorization.AuthorizationClientExtensions.ToPSRoleAssignments(IEnumerable`1 assignments, List`1 roleDefinitions, AuthorizationClient policyClient, ActiveDirectoryClient activeDirectoryClient, Boolean excludeAssignmentsForDeletedPrincipals)
2017-01-18T17:43:56.2569844Z ##[debug]   at Microsoft.Azure.Commands.Resources.Models.Authorization.AuthorizationClient.FilterRoleAssignments(FilterRoleAssignmentsOptions options, String currentSubscription)
2017-01-18T17:43:56.2579843Z ##[debug]   at Microsoft.Azure.Commands.Resources.GetAzureRoleAssignmentCommand.ExecuteCmdlet()
2017-01-18T17:43:56.2599842Z ##[debug]   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
2017-01-18T17:43:56.2959008Z ##[error]Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
2017-01-18T17:43:56.2959008Z ##[debug]Processed: ##vso[task.logissue type=error]Exception of type 'Microsoft.Rest.Azure.CloudException' was thrown.
2017-01-18T17:43:56.2989006Z ##[debug]Processed: ##vso[task.complete result=Failed]
2017-01-18T17:43:56.3229004Z ##[section]Finishing: Azure PowerShell script: InlineScript

### PR DESCRIPTION
**Pull request by darshanhs90**
This pr is as part of the following bug fix
#3407

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by darshanhs90** [comment]
@cormacpayne the build is failing due to some change unrelated to my code.Does this block the merging of my Pull request?

**Comment by cormacpayne** [comment]
@darshanhs90 please assign back to me and add the `needs-review` label when changes to other projects (updating to 4.5.2) are pushed

**Comment by darshanhs90** [comment]
@cormacpayne updated the dependencies and assigned back to you with the needs-review label

**Comment by markcowl** [comment]
@darshanhs90 Tere are merge conflicts, please pull the latest from the preview branch and update.  Are you adding any dependencies with new filenames?  If not, you do not need to update the wxi file.

**Comment by markcowl** [comment]
on demand run (just to check issues) here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1505/

**Comment by darshanhs90** [comment]
@markcowl the changes done are part of the dependencies where the earlier projects used reference like net45/graph.rbac.dll but the new one uses reference like net452/graph.rbac.dll.
Since this resulted in an build error due to wxi file not being updated.
As per the advice from @cormacpayne I had to regenerate the wxi file and include that also

**Comment by darshanhs90** [comment]
@markcowl updated the wxi files and fixed the build issues

**Comment by markcowl** [comment]
You should not need to update the wxi at all for this change.  Please checkout the version upstream and resubmit.  The changes to the wxi will not work

**Comment by darshanhs90** [comment]
@markcowl when i remove the wxi file and revert to the upstream version,I am getting build issues as per this log http://azuresdkci.cloudapp.net/job/powershell/8455/console 
Is there something that i am missing which is causing this issue in the build?

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -7554,4 +7554,4 @@
             <ComponentRef Id="cmpE6E248F4029BF7DACCD9C87C2D38BD38" />
         </ComponentGroup>
     </Fragment>
-</Include>
+</Include>
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: Revert any change here - this is likely causing your build issues.  Please check out the latest version of the wxi file.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: setup/azurecmdfiles.wxi (hunk 1 above)
Comment (darshanhs90) [inline setup/azurecmdfiles.wxi]: @markcowl checked out the wxi file ,got the build error again

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1521/

**Comment by markcowl** [comment]
@darshanhs90 inside your powershell clone, run this command:

```
git checkout -f upstream/preview -- setup/azurecmdfiles.wxi
```

This will checkout the version of the wxi file in the preview branch.  Then push to your branch.  There should be no change to the wxi file in your PR.  Please ensure that your PR does not show any change to the wxi file.

**Comment by darshanhs90** [comment]
@markcowl updated the wxi file using the command mentioned by you

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1524/

**Comment by markcowl** [comment]
@darshanhs90 This is still failing, it looks like your local clone may not have been up to date when you did this, please try git fetch upstream and repeat the command above.

If not, I will recreate a new PR for you with these changes tomorrow.

**Comment by markcowl** [comment]
@darshanhs90 We opened this PR: #3892 which has the same changes as this one.  Closing in favor of the new PR

**Pull request closed without merging**